.github/workflows/docs-build-push.yml

```diff
@@ -0,0 +1,36 @@
+name: Build and deploy docs
+on:
+  workflow_dispatch:
+    inputs:
+      environment:
+        description: 'Environment to deploy to'
+        required: true
+        default: 'preview'
+        type: choice
+        options:
+        - preview
+        - dev
+        - staging
+        - prod
+  pull_request:
+    branches:
+      - "*"
```

Comment:
Where will this publish to? Can we control which branches publish where?

For release branches (`release-X.X`), we want the docs published to prod. Every other branch should only publish previews, not prod.

Reply:
When a PR is created that changes docs, a preview deploy will be made, and a URL will be added as a comment.
This _does not_ deploy automatically, anywhere right now. You have to manually trigger the action, pick your branch, and pick your environment.
 We're working on an update this week to allow users to specific a branch to autodeploy to an environment - I can hold off on this PR until that's ready if you like.

Reply:
Oh no that's fine, the preview on PRs is all that's needed. I think the only time that we want to actually publish is when docs are pushed to a release branch. 

In netlify, we would first publish from the release branch after we pointed netlify at the branch (manual step that we did once the release was out), and then any docs changes in that branch going forward would be deployed automatically. I'm guessing that's the process you're referring to that you're working on?

Sidenote: it would be nice if we could have the doc preview exist in the actions list at the bottom of the PR instead of as a comment (just to save on the PR notifications and clutter in the comment section).

Reply:
Yes, exactly what we're working on at the moment 😄 . We're trying to mimic your old workflow from Netlify.

I like the idea of adding the url to the bottom of the PR, would need some extra parsing logic, but doable for sure 👍 .